zebra-rpc/src/methods/tests/vectors.rs

```diff
@@ -1791,44 +1796,39 @@ async fn rpc_getnetworksolps() {
 }
 
 #[tokio::test(flavor = "multi_thread")]
-async fn rpc_getblocktemplate() {
-    // test getblocktemplate with a miner P2SH address
-    rpc_getblocktemplate_mining_address(true).await;
-    // test getblocktemplate with a miner P2PKH address
-    rpc_getblocktemplate_mining_address(false).await;
-}
-
-async fn rpc_getblocktemplate_mining_address(use_p2pkh: bool) {
+async fn getblocktemplate() {
     let _init_guard = zebra_test::init();
 
+    // TODO Run the test with all possible network types.
+    let net = Network::Mainnet;
+
+    // TODO Run the test with all address types supported for mining.
```

Comment:
Another very nice refactor, thank you. 

Are these TODOs meant to be resolved in this PR (or a follow-up PR)?

Nitpick:

```suggestion
    // TODO: Run the test with all possible network types.
    let net = Network::Mainnet;

    // TODO: Run the test with all address types supported for mining.
```

Reply:
I was going to address them once we can mine to the shielded addresses.